Onboarding Feature

baskurthalit opened the pull request:
### Task Link :link:
[Task](https://github.com/orgs/SwiftBuddiesTR/projects/3?pane=issue&itemId=57480927)
<!-- What is your redmine link for this task? -->

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->
Develop OnboardingModule for the project.

### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->
Created a module called OnboardingModule.
Developed OnboardingView with two slides.

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->
Checked in different screen sizes.

### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->
<img src="https://github.com/SwiftBuddiesTR/BuddiesIOS/assets/57860204/b2d65d6b-a09e-415c-a998-9afab05f76fe" width="150">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

baskurthalit committed:
Onboarding feature developed.

canyoldas0 commented on SwiftBuddiesIOS/Targets/DesignModule/Sources/ViewComponents/Shapes/RoundedCorner.swift:
```diff
@@ -0,0 +1,28 @@
+}
+
+public extension View {
+    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
```
Custom bir implementation'a ihtiyac oldugunu sanmiyorum iOS 16 ile birlikte

[Ornek](https://stackoverflow.com/a/58606176/14868085)

canyoldas0 commented on SwiftBuddiesIOS/Targets/Onboardingmodule/Sources/Presentation/OnbordingScreen/OnboardingView.swift:
```diff
@@ -0,0 +1,64 @@
+    
+    func setPageIndicatorColor() {
+        UIPageControl
+            .appearance()
```
`UIPageControl.appearance()` i degistirdigin zaman uygulama genelinde tum UIPageControl'lerin appearance'ini degisiyor, suan icin bence bir sorun degil ama aklinda olsun 👍

canyoldas0 commented on SwiftBuddiesIOS/Sources/ContentView.swift:
```diff
@@ -1,12 +1,53 @@
 
 public struct ContentView: View {
-    public init() {}
+    @State var isOnboardingSeen: Bool
```
`@AppStorage` ile degistirilebilir.

baskurthalit committed:
Custom corner radius shape deleted.

baskurthalit committed:
Merge branch 'dev' into feature-Onboarding

baskurthalit committed:
Some logics for OnboardingView in main target are moved module itself.
LocalDataManager deleted.
IsSeenOnboarding is handled by @AppStorage()
Remove unnecessary lines.

baskurthalit committed:
Remove unnecessary default values.

baskurthalit commented on SwiftBuddiesIOS/Sources/ContentView.swift:
```diff
@@ -1,12 +1,53 @@
+    }
+    
+    private func prepareOnboardingItems() -> [OnboardingItemModel] {
+        [.init(id: 0,
```
Onboarding Module içerisine taşıdım.

baskurthalit commented on SwiftBuddiesIOS/Sources/ContentView.swift:
```diff
@@ -1,12 +1,53 @@
+    
+    private func onboardingSeenAction() {
+        withAnimation(.easeInOut) { isOnboardingSeen = true }
+        LocalData.manager.isOnboardingScreenViewed = isOnboardingSeen
```
Mantıklı evet, düzelttim.

baskurthalit commented on SwiftBuddiesIOS/Sources/Managers/LocalData.swift:
```diff
@@ -0,0 +1,15 @@
+
+import SwiftUI
+
+final class LocalData {
```
LocalData'ı sildim komple. @AppStorage'e bağlı çalışıyor.

baskurthalit committed:
Default func parameters updated.

baskurthalit commented on SwiftBuddiesIOS/Targets/DesignModule/Sources/Extensions/View+Extensions.swift:
```diff
@@ -45,4 +45,11 @@ public extension View {
         }
     }
+    
+    func fillView(_ color: Color) -> some View {
```
Güncelledim.

baskurthalit committed:
Merge branch 'dev' into feature-Onboarding

# Conflicts:
#	SwiftBuddiesIOS/Sources/ContentView.swift

canyoldas0 reviewed (approved):
Eline saglik 👍
